conda/core/repodata.py

```diff
@@ -74,6 +74,24 @@ def __call__(cls, channel):
 class SubdirData(object):
     _cache_ = {}
 
+    @staticmethod
+    def reverse_query_all(channels, subdirs, package_ref_spec):
+        from .index import check_whitelist  # TODO: fix in-line import
+        channel_urls = all_channel_urls(channels, subdirs=subdirs)
+        check_whitelist(channel_urls)
+        with ThreadLimitedThreadPoolExecutor() as executor:
+            futures = (executor.submit(
+                SubdirData(Channel(url)).reverse_query, package_ref_spec
+            ) for url in channel_urls)
+            return tuple(concat(future.result() for future in as_completed(futures)))
+
+    def reverse_query(self, package_ref_spec):
```

Comment:
Let's call `package_ref_spec ` instead `package_ref_or_match_spec` for consistency.  `query()` is intended to be an API-level function, so we should get this right here.